pkg/crds/crds.go

```diff
@@ -0,0 +1,160 @@
+	"errors"
+	"fmt"
+	"github.com/rook/kubectl-rook-ceph/pkg/k8sutil"
+	"github.com/rook/kubectl-rook-ceph/pkg/logging"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/types"
+	"strings"
+)
+
+var cephResources = []string{
+	"cephblockpoolradosnamespaces",
+	"cephblockpools",
+	"cephbucketnotifications",
+	"cephbuckettopics",
+	"cephclients",
+	"cephclusters",
+	"cephcosidrivers",
+	"cephfilesystemmirrors",
+	"cephfilesystems",
+	"cephfilesystemsubvolumegroup",
+	"cephnfses",
+	"cephobjectrealms",
+	"cephobjectstores",
+	"cephobjectstoreusers",
+	"cephobjectzonegroups",
+	"cephobjectzones",
+	"cephrbdmirrors",
+}
+
+const (
+	cephRookIoGroup          = "ceph.rook.io"
+	cephRookResourcesVersion = "v1"
+)
+
+const (
+	CephResourceKindCephCluster = "CephCluster"
+)
+
+var (
+	errorResourceNotFound         = errors.New("the server could not find the requested resource")
+	clusterResourcePatchFinalizer = map[string]interface{}{
+		"spec": map[string]interface{}{
+			"cleanupPolicy": map[string]string{
+				"confirmation": "yes-really-destroy-data",
+			},
+		},
+	}
+)
+
+func DeleteCustomResources(ctx context.Context, clientsets k8sutil.ClientsetsInterface, clusterNamespace string) {
+	err := deleteCustomResources(ctx, clientsets, clusterNamespace)
+	if err != nil {
+		logging.Fatal(err)
+	}
+	logging.Info("done")
+}
+
+func deleteCustomResources(ctx context.Context, clientsets k8sutil.ClientsetsInterface, clusterNamespace string) error {
+	for _, resource := range cephResources {
+		logging.Info(fmt.Sprintf("Getting resources kind %s", resource))
+		items, err := clientsets.ListResourcesDynamically(ctx, cephRookIoGroup, cephRookResourcesVersion, resource, clusterNamespace)
+		if err != nil {
+			if strings.Contains(err.Error(), errorResourceNotFound.Error()) {
+				logging.Warning("the server could not find the requested resource: %s", resource)
+				continue
+			}
+			// log instead of return
+			return err
+		}
+
+		if len(items) == 0 {
+			logging.Warning("resource %s was not found on the cluster", resource)
+			continue
+		}
+
+		for _, item := range items {
+			logging.Info(fmt.Sprintf("removing resource %s: %s", resource, item.GetName()))
+			err = clientsets.DeleteResourcesDynamically(ctx, cephRookIoGroup, cephRookResourcesVersion, resource, clusterNamespace, item.GetName())
+			if err != nil {
+				if err.Error() == "the server could not find the requested resource" {
+					logging.Info(err.Error())
+					continue
+				}
+				// log instead of return
+				return err
+			}
+
+			itemResource, err := clientsets.GetResourcesDynamically(ctx, cephRookIoGroup, cephRookResourcesVersion, resource, item.GetName(), clusterNamespace)
+			if err != nil {
+				if err.Error() == "the server could not find the requested resource" {
+					logging.Info(err.Error())
+					continue
+				}
+				// log instead of return
+				return err
+			}
+
+			if itemResource != nil {
+				logging.Info("resource still alive, applying patch...")
```

Comment:
With each log message, let's include details to help understand what is happening.
```suggestion
				logging.Infof("resource %q is not yet deleted, applying patch to remove finalizer...", itemResource.Name)
```